src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Entity/SaveOidcngEntityCommandHandler.php

```diff
@@ -96,8 +96,13 @@ public function handle(SaveOidcngEntityCommand $command)
         $entity->setAccessTokenValidity($command->getAccessTokenValidity());
         $entity->setIsPublicClient($command->isPublicClient());
         $entity->setEnablePlayground($command->isEnablePlayground());
-        $grantType =  $command->getGrantType();
-        $entity->setGrantType(new OidcGrantType($grantType));
+        try {
+            $grantType =  $command->getGrantType();
+            $entity->setGrantType(new OidcGrantType($grantType));
+        } catch (\InvalidArgumentException $e) {
```

Comment:
Yes, in draft state, the grant type can be unset and the validation is not triggered. In all other situations the grant type must be set, so refactoring the code to allow not setting is would not be a good idea.

Reply:
Discussed IRL and I thought there should be a default state selected. Then I think this would be sufficient for now. I'll add this to the draft removal issue so we won't forget to remove this then.

Reply:
https://www.pivotaltracker.com/story/show/166004131 (task)